### PR TITLE
#684 Relative Symlinks

### PR DESCRIPTION
**Pull request by dwkitchen**
Based on Symfony/Filesystem->makePathRelative()

### CODE REVIEW

File: classes/phing/tasks/ext/SymlinkTask.php [hunk 1]
```diff
@@ -236,7 +236,36 @@ protected function getMap()
             }
 
             if ($this->isRelative()) {
-                $fromDir = $fs->getDir($this->getProject())->getPath();
+                $endPath = $fs->getDir($this->getProject())->getAbsolutePath();
+                $startPath = $this->getLink();
```
Comment (dwkitchen) [inline classes/phing/tasks/ext/SymlinkTask.php]: This only works if the link path given is absolute, how to get absolute path from `$this->getLink()?

Comment (siad007) [inline classes/phing/tasks/ext/SymlinkTask.php]: You can get the absolute path for `$this->getLink()` in combination with `PhingFile` like:
```
$startPath = (new PhingFile($this->getLink()))->getAbsolutePath();
```

Comment (dwkitchen) [inline classes/phing/tasks/ext/SymlinkTask.php]: @siad007 Thanks for that updated and working with a relative path to the project root is given in the command.

**Comment by mrook** [comment]
Closed in favor of #695 - thanks for your contribution!

**Pull request closed without merging**